### ISSUE TITLE
jsb.onError design issues

### ISSUE THREAD

**Issue by smallmain** [cocos/cocos-engine#17025]
### Cocos Creator version

3.8.2

### System information

MacOS

### Issue description

- The current `jsb.onError` will capture all errors. It is recommended that the design be similar to Web and NodeJS, and onError should be divided into at least two different events: `onUnhandledRejection` and `onError`.
- `jsb` is now marked as a deprecated interface, but there is no replacement interface.
- Should be used as a platform layer for cross-platform abstraction, for example: `sys.getNetworkType()` api.

### Steps to reproduce

.

**Comment by dumganhar** [comment]
`jsb` is replaced with `native`, I think we need to put `onError` to `native` namespace.

Yes, jsb.onError capture all errors which is not as standard as which in web and nodejs.

https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event
https://www.geeksforgeeks.org/node-js-process-unhandledpromiserejection-event/

Promise unhandled rejection should be captured by a separate callback.

**Comment by smallmain** [comment]
> `jsb` is replaced with `native`, I think we need to put `onError` to `native` namespace.
> 
> Yes, jsb.onError capture all errors which is not as standard as which in web and nodejs.
> 
> https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event https://www.geeksforgeeks.org/node-js-process-unhandledpromiserejection-event/
> 
> Promise unhandled rejection should be captured by a separate callback.

I don't have the same idea as you. I think it should be placed in an interface like `sys.on("error")` to adapt to all platforms, so that users can use this interface across platforms.

**Comment by smallmain** [comment]
Because if users want to catch errors, no one wants to only take effect on the native side, so they have to adapt themselves.

**Comment by dumganhar** [comment]
Agree. It seems `sys.on('error', function() {...})` be a better choise.

**Comment by dumganhar** [comment]
This issue has been delayed. Put it to next milestone.

**Comment by dumganhar** [comment]
It needs to be delayed again.
